frontend/src/app/permission.service.ts

```diff
@@ -7,12 +7,22 @@ import { AdminSettingsNavigationData } from './navigation/navigation.service';
   providedIn: 'root'
 })
 export class PermissionService {
+  private profile: Signal<Profile | undefined> = signal(undefined);
   private profile$: Observable<Profile | undefined>;
 
   constructor(profileService: ProfileService) {
+    this.profile = profileService.profile;
     this.profile$ = profileService.profile$;
   }
 
+  checkSignal(action: string, resource: string): boolean {
+    if (this.profile() === undefined) {
+      return false;
+    } else {
+      return this.hasPermission(this.profile()!.permissions, action, resource);
+    }
+  }
```

Comment:
I would probably flip this logic to let the truthy case come first with just : `if (this.profile())`

Reply:
Collapsed this down even further @KrisJordan, but let me know if the other version is preferred for readability!